ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/events/filesystem/FSTableEvent.java

```diff
@@ -0,0 +1,106 @@
+package org.apache.hadoop.hive.ql.exec.repl.bootstrap.events.filesystem;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.Warehouse;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.ql.exec.repl.bootstrap.events.TableEvent;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.parse.EximUtil;
+import org.apache.hadoop.hive.ql.parse.ReplicationSpec;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.parse.repl.load.MetaData;
+import org.apache.hadoop.hive.ql.plan.AddPartitionDesc;
+import org.apache.hadoop.hive.ql.plan.ImportTableDesc;
+import org.apache.hadoop.hive.ql.plan.PlanUtils;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FSTableEvent implements TableEvent {
+  private final Path fromPath;
+  private final MetaData metadata;
+
+  FSTableEvent(HiveConf hiveConf, String metadataDir) {
+    try {
+      URI fromURI = EximUtil.getValidatedURI(hiveConf, PlanUtils.stripQuotes(metadataDir));
+      fromPath = new Path(fromURI.getScheme(), fromURI.getAuthority(), fromURI.getPath());
+      FileSystem fs = FileSystem.get(fromURI, hiveConf);
+      metadata = EximUtil.readMetaData(fs, new Path(fromPath, EximUtil.METADATA_NAME));
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public boolean shouldNotReplicate() {
+    ReplicationSpec spec = metadata.getReplicationSpec();
+    return spec.isNoop() || !spec.isInReplicationScope();
+  }
+
+  @Override
+  public Path metadataPath() {
+    return fromPath;
+  }
+
+  @Override
+  public ImportTableDesc tableDesc(String dbName) throws SemanticException {
+    try {
+      Table table = new Table(metadata.getTable());
+      ImportTableDesc tableDesc = new ImportTableDesc(dbName, table);
+      tableDesc.setReplicationSpec(metadata.getReplicationSpec());
+      return tableDesc;
+    } catch (Exception e) {
+      throw new SemanticException(e);
+    }
+  }
+
+  @Override
+  public List<AddPartitionDesc> partitionDescriptions(ImportTableDesc tblDesc)
```

Comment:
we can though the whole list itself can be big for large number of partitions in a table. eventual plan is to solve all of this by implementing HIVE-16904